### DISCUSSION TITLE
bring back Python 3.7 support.

### DISCUSSION THREAD

**Discussion by T-256** [frankie567/httpx-ws#41] [General]
`httpx` still didn't drop it.
In Python 3.7 I can receive latest `httpcore 0.17.3` but the latest `httpx-ws 0.3.1` has problem with it (https://github.com/frankie567/httpx-ws/issues/35).
I need `httpcore` latest bugfixes so I can't pin it to compatible version for `httpx-ws`.
Currently I most use [my fork](https://github.com/T-256/httpx-ws), But if you want, we can push it to py37 branch? at least until httpx drop it.

**Comment by frankie567** [comment]
To be quite frank, no. Python 3.7 is now unsupported and put applications at risk. Besides, I can't afford personally to maintain a dedicated branch for compatibility.

I don't know what your requirements/constraints are, but I would say that you should try to upgrade to Python 3.8. Most likely, it won't break anything in your code.

**Reply by T-256** [reply]
My project based on Python 3.7 AST instructions (compiled codes e.g. `.pyc` files), I cannot parse Python 3.7 AST in Python 3.8.
